### DISCUSSION TITLE
speed up large data set

### DISCUSSION THREAD

**Discussion by teathinker** [epezent/implot#342] [Ideas]
Hi, first thanks for this excellent framework.

I am testing on some big dataset and find that it could be slow and lagging even if I zoom into details with only a small subset of data in view. So I am wondering:

Is there any way to only draw&render the small subset automatically when zoomed in?
Is there any way to automatically merge/resample data so that we can get better performance when zoomed out?
Thanks!

**Comment by epezent** [comment]
Great! Consider sharing your solution with others.

**Reply by teathinker** [reply]
No problem. I just resumed the question. Solution is straight forward:
1. I just get the Range of the axis and only prepare data to render in that range.
2. There isn't an automatical way since different data needs different way to merge. I have to implement it as in the first solution above.
